src/include/miopen/nllloss/problem_description.hpp

```diff
@@ -0,0 +1,261 @@
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetTargetDesc() const { return targetDesc; }
+    const TensorDescriptor& GetWeightDesc() const { return weightDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+    int32_t GetIgnoreIndex() const { return ignore_index; }
+
+    bool IsRightLength() const
+    {
+        if(targetDesc.GetLengths()[0] != inputDesc.GetLengths()[0])
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "NLLLoss: Tensor sizes do not match.");
+#else
+            return false;
+#endif
+        for(int32_t i = 1; i < targetDesc.GetSize(); ++i)
+        {
+            if(targetDesc.GetLengths()[i] != inputDesc.GetLengths()[i + 1])
+            {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+                MIOPEN_THROW(miopenStatusBadParm, "NLLLoss: Tensor sizes do not match.");
+#else
+                return false;
+#endif
+            }
+        }
+        if(weightDesc.GetLengths()[0] != inputDesc.GetLengths()[1])
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "NLLLoss: Tensor sizes do not match.");
+#else
+            return false;
+#endif
+        }
+        return true;
+    }
+
+    bool IsRightStride() const
+    {
+        auto isRightStride = [](TensorDescriptor td) {
+            auto strides = td.GetStrides();
+            auto lengths = td.GetLengths();
+            std::vector<std::pair<size_t, size_t>> p;
+            p.reserve(td.GetSize());
+            std::transform(strides.begin(),
+                           strides.end(),
+                           lengths.begin(),
+                           std::back_inserter(p),
+                           [](size_t a, size_t b) { return std::make_pair(a, b); });
+            std::sort(p.begin(), p.end());
+            for(int i = 1; i < p.size(); ++i)
+            {
+                if(p[i].first != p[i - 1].first * p[i - 1].second)
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+                    MIOPEN_THROW(miopenStatusBadParm, "NLLLoss: Tensor strides do not match.");
+#else
+                    return false;
+#endif
+            }
+            return true;
+        };
+        return isRightStride(inputDesc) && isRightStride(targetDesc) && isRightStride(outputDesc) &&
+               isRightStride(weightDesc);
+    }
+
+    bool IsAllContiguous() const
+    {
+        auto isContiguous = [](TensorDescriptor td) {
+            size_t s = 1;
+            for(int i = td.GetSize() - 1; i >= 0; --i)
+            {
+                if(s != td.GetStrides()[i])
+                {
+                    return false;
+                }
+                s *= td.GetLengths()[i];
+            }
+            return true;
+        };
+        return isContiguous(inputDesc) && isContiguous(targetDesc) && isContiguous(weightDesc) &&
+               isContiguous(outputDesc);
+    }
+
+    bool IsAllPacked() const
```

Comment:
Please refer to [slack ](https://morehvietnam.slack.com/archives/C06RZ10CKAA/p1715667533752279)and add a function to check contiguous.